File: natour-backend/src/main/java/com/natour/natour/model/Stars.java
package com.natour.natour.model;

public enum Stars {
    ZERO, ONE, TWO, THREE, FOUR, FIVE;

    public static Stars getStarsFromAvgReviews(double avg) {
        if (avg < 0.5) return ZERO;
        if (avg < 1.5) return ONE;
        if (avg < 2.5) return TWO;
        if (avg < 3.5) return THREE;
        if (avg < 4.5) return FOUR;
        if (avg <= 5) return FIVE;
        else throw new IllegalArgumentException("Illegal avg reviews");
    }
}

File: natour-android/app/src/main/java/com/example/natour/data/model/Stars.kt
package com.example.natour.data.model

enum class Stars {
    ZERO, ONE, TWO, THREE, FOUR, FIVE;

    companion object {
        fun intToEnumValue(int: Int): Stars = when(int) {
            0 -> ZERO
            1 -> ONE
            2 -> TWO
            3 -> THREE
            4 -> FOUR
            5 -> FIVE
            else -> throw IllegalArgumentException("Unknown number stars enum value")
        }
    }
}